src/distance.hpp

```diff
@@ -0,0 +1,33 @@
+#include "snarls.hpp"
+
+namespace vg {
+    
+    struct ChainDistances {
+        /*Stores distance information for snarls in a chain*/
+        unordered_map<id_t, size_t>* snarl_to_index;
+        vector<int64_t>* prefix_sum;//Dist from start of chain to each snarl
+    };
+
+
+    struct SnarlDistances {
+        
+        /* Stores distance information for nodes in a snarl.
+        visit_to_index maps each visit (node_id, negative if reverse) to an int
+        distances stores all the distance bewteen each pair of visits in a snarl
+        */
+        unordered_map<pair<id_t, bool>, size_t>* visit_to_index;
+        vector<int64_t>* distances;
+    };
+
+
+    struct DistanceIndex {
+
+        /*The distance index containing snarl and chain information*/
+        unordered_map<id_t, SnarlDistances*> sd;
+             //map from node id of first nodein snarl
+        unordered_map<id_t, ChainDistances*> cd;
+    };
+      
+    
+    DistanceIndex makeDistanceIndex(VG* graph, SnarlManager& sm);
+}
```

Comment:
You'll need to have forward declarations for the functions in the .hpp file too, or else no other part of the program will be able to use them. The .cpp files don't get included by other files, so anything that's defined only in the .cpp file only lives in the .cpp file. Ideally, the functions should migrate into classes as methods, so the forward declarations would go along with the class definition.